rest_framework_json_api/schemas/openapi.py

```diff
@@ -0,0 +1,878 @@
+            return 'list'
+        else:
+            return 'fetch'
+
+
+class AutoSchema(drf_openapi.AutoSchema):
+    """
+    Extend DRF's openapi.AutoSchema for JSONAPI serialization.
+    """
+    #: ignore all the media types and only generate a JSONAPI schema.
+    content_types = ['application/vnd.api+json']
+
+    def get_operation(self, path, method, action=None):
+        """
+        JSONAPI adds some standard fields to the API response that are not in upstream DRF:
+        - some that only apply to GET/HEAD methods.
+        - collections
+        - special handling for POST, PATCH, DELETE:
+
+        :param action: One of the usual actions for a conventional path (list, retrieve, update,
+            partial_update, destroy) or special case 'rel' or 'rels' for a singular or
+            plural relationship.
+        """
+        operation = {}
+        operation['operationId'] = self.get_operation_id(path, method)
+        operation['description'] = self.get_description(path, method)
+        if hasattr(self, 'get_security_requirements'):  # pragma: no cover
+            security = self.get_security_requirements(path, method)
+            if security is not None:
+                operation['security'] = security
+
+        parameters = []
+        parameters += self.get_path_parameters(path, method)
+        # pagination, filters only apply to GET/HEAD of collections and items
+        if method in ['GET', 'HEAD']:
+            parameters += self._get_include_parameters(path, method)
+            parameters += self._get_fields_parameters(path, method)
+            parameters += self._get_sort_parameters(path, method)
+            parameters += self.get_pagination_parameters(path, method)
+            parameters += self.get_filter_parameters(path, method)
+        operation['parameters'] = parameters
+
+        # get request and response code schemas
+        if method == 'GET':
+            if is_list_view(path, method, self.view):
+                self._get_collection_response(operation)
+            else:
+                self._get_item_response(operation)
+        elif method == 'POST':
+            self._post_item_response(operation, path, action)
+        elif method == 'PATCH':
+            self._patch_item_response(operation, path, action)
+        elif method == 'DELETE':
+            # should only allow deleting a resource, not a collection
+            # TODO: implement delete of a relationship in future release.
+            self._delete_item_response(operation, path, action)
+        return operation
+
+    def get_operation_id(self, path, method):
+        """
+        The upstream DRF version creates non-unique operationIDs, because the same view is
+        used for the main path as well as such as related and relationships.
+        This concatenates the (mapped) method name and path as the spec allows most any
+        """
+        method_name = getattr(self.view, 'action', method.lower())
```

Comment:
Hmm, it's a toss-up. I supposed I could change the base depending on the particular path (main/related/relationships) but is it necessary? Why not just use the path? The allowable character set for operation IDs is "string".

Reply:
I am not suggesting to change the logic but to overwrite less logic of DRF. At a quick glance it seems that simply overwriting `get_operation_id_base` like below instead of `get_operation_id` would lead to the same result as now. 

```python
def get_operation_id_base(self, path, method, action):
   return path
```

Reply:
Oh, I see. I misread the recently-added description of get_operation_id_base. I thought it was a class attribute not a method. It was added in 3.12. I suppose I can do that. Will take a deeper look.

Reply:
Actually operation_id_base is an Autoschema parameter - so that is the static part I was thinking of. get_operation_id_base is *much* longer than get_operation_id and the net result is I would have to do extra work to determine if the path is for the base collection/item or for the related or relationships.... Not sure it's worth doing that just to get a unique operationID that nobody even cares about but is required by the OAS spec.

I stared at this for about an hour and it's harder than it appears. There are larger unsolved issues I'd like to focus on for this initial release like fixing `_expand_relationships()`